manifests/renew.pp

```diff
@@ -26,16 +26,20 @@
 # @param cron_monthday
 #   Optional string, integer or array of monthday(s) the renewal command should
 #   run. E.g. '2-30/2' to run on even days. Default: Every day.
+# @param cron_environment
+#   Optional string or array of environments(s) the renewal command should have.
+#   E.g. PATH=/sbin:/usr/sbin:/bin:/usr/bin
 #
 class letsencrypt::renew (
-  Variant[String[1], Array[String[1]]] $pre_hook_commands    = $letsencrypt::renew_pre_hook_commands,
-  Variant[String[1], Array[String[1]]] $post_hook_commands   = $letsencrypt::renew_post_hook_commands,
-  Variant[String[1], Array[String[1]]] $deploy_hook_commands = $letsencrypt::renew_deploy_hook_commands,
-  Array[String[1]]                     $additional_args      = $letsencrypt::renew_additional_args,
-  Enum['present', 'absent']            $cron_ensure          = $letsencrypt::renew_cron_ensure,
-  Letsencrypt::Cron::Hour              $cron_hour            = $letsencrypt::renew_cron_hour,
-  Letsencrypt::Cron::Minute            $cron_minute          = $letsencrypt::renew_cron_minute,
-  Letsencrypt::Cron::Monthday          $cron_monthday        = $letsencrypt::renew_cron_monthday,
+  Variant[String[1], Array[String[1]]]  $pre_hook_commands    = $letsencrypt::renew_pre_hook_commands,
+  Variant[String[1], Array[String[1]]]  $post_hook_commands   = $letsencrypt::renew_post_hook_commands,
+  Variant[String[1], Array[String[1]]]  $deploy_hook_commands = $letsencrypt::renew_deploy_hook_commands,
+  Array[String[1]]                      $additional_args      = $letsencrypt::renew_additional_args,
+  Enum['present', 'absent']             $cron_ensure          = $letsencrypt::renew_cron_ensure,
+  Letsencrypt::Cron::Hour               $cron_hour            = $letsencrypt::renew_cron_hour,
+  Letsencrypt::Cron::Minute             $cron_minute          = $letsencrypt::renew_cron_minute,
+  Letsencrypt::Cron::Monthday           $cron_monthday        = $letsencrypt::renew_cron_monthday,
+  Variant[String, Array[String], Undef] $cron_environment     = $letsencrypt::renew_cron_environment,
```

Comment:
Would it make sense for the type to default to an empty array and disallow `Undef`? I'm not sure if that makes a difference for the cron type.

Reply:
I've tested with [] and it removed the environment if it was previously setted. With the option unset (or undef) it doesn't remove the environment. But I think that removing the environment might be the right thing to do.

Reply:
> I've tested with [] and it removed the environment if it was previously setted. With the option unset (or undef) it doesn't remove the environment. But I think that removing the environment might be the right thing to do.

@gmenuel I wonder then if we should mark this as a breaking change, since it would remove manual changes to the cron job.

Reply:
@kenyon yes you are right, it is probably better to tag this as a breaking change.